sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/TransposeWindowSuite.scala

```diff
@@ -142,4 +142,20 @@ class TransposeWindowSuite extends PlanTest {
     comparePlans(optimized, analyzed)
   }
 
+  test("transpose two adjacent windows with compatible partitions which is not a prefix") {
```

Comment:
quick question, did this fail before? I thought that PR proposes a perf improvement.

Reply:
This will fail before. 
This PR has two effects for the rule: 1. extend the applicable cases(like the case added in the UT) 2. perf improvement